stan/math/prim/fun/quantile.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef STAN_MATH_PRIM_FUN_QUANTILE_HPP
+#define STAN_MATH_PRIM_FUN_QUANTILE_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/as_array_or_scalar.hpp>
+#include <algorithm>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return sample quantiles corresponding to the given probabilities.
+ * The smallest observation corresponds to a probability of 0 and the largest to
+ * a probability of 1.
+ *
+ * Implementation follows the default R behavior, as defined here:
+ * https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/quantile
+ *
+ * @tparam T Type of elements contained in vector.
+ * @param xs Numeric vector whose sample quantiles are wanted
+ * @param p Probability with value between 0 and 1.
+ * @return Sample quantile.
+ * @throw std::invalid_argument If any of the values are NaN or size 0.
```

Comment:
If any element of xs is NaN. I guess if p was NaN it wouldn't be invalid_argument.